docs/network protocol.md

```diff
@@ -646,15 +645,48 @@ class Hint(typing.NamedTuple):
 ```
 
 ### Data Package Contents
-A data package is a JSON object which may contain arbitrary metadata to enable a client to interact with the Archipelago server most easily. Currently, this package is used to send ID to name mappings so that clients need not maintain their own mappings.
-
-We encourage clients to cache the data package they receive on disk, or otherwise not tied to a session. You will know when your cache is outdated if the [RoomInfo](#RoomInfo) packet or the datapackage itself denote a different version. A special case is datapackage version 0, where it is expected the package is custom and should not be cached.
-
-Note: 
- * Any ID is unique to its type across AP: Item 56 only exists once and Location 56 only exists once.
- * Any Name is unique to its type across its own Game only: Single Arrow can exist in two games.
- * The IDs from the game "Archipelago" may be used in any other game. 
-   Especially Location ID -1: Cheat Console and -2: Server (typically Remote Start Inventory)
+A data package is a JSON object which may contain arbitrary metadata to enable a client to interact with the Archipelago
+server most easily and not maintain their own mappings. Some contents include:
+
+   - Name to ID mappings for items and locations.
+   - A checksum of each game's data package for clients to tell if a cached package is invalid.
+
+We encourage clients to cache the data package they receive on disk, or otherwise not tied to a session. You will know 
+when your cache is outdated if the [RoomInfo](#RoomInfo) packet or the datapackage itself denote a different checksum
+than any locally cached ones.
+
+**Important Notes about IDs and Names**: 
+
+* IDs ≤ 0 are reserved for "Archipelago" and should not be used by other world implementations.
+* The IDs from the game "Archipelago" (in `worlds/generic`) may be used in any world.
+  * Especially Location ID `-1`: `Cheat Console` and `-2`: `Server` (typically Remote Start Inventory)
+* Any names and IDs are only unique in its own world data package, but different games may reuse these names or IDs.
+  * At runtime, you will need to look up the game of the player to know which item or location ID/Name to lookup in the
+    data package. This can be easily achieved by reviewing the `slot_info` for a particular player ID prior to lookup.
+  * For example, a data package like this is valid:
+    ```jsonc
```

Comment:
Using jsonc in docs might make someone believe their client can end up receiving jsonc. Might be dangerous?

Reply:
I think I just used it for comment highlighting. I'll add a clarification saying that comments are not sent in the actual JSON.

Reply:
Actually just moved the comment out of the block entirely.